instance/ansible.py

```diff
@@ -67,59 +68,96 @@ def dict_merge(dict1, dict2):
     return dict1
 
 
-@contextmanager
-def string_to_file_path(string):
+def string_to_file_path(string, root_dir=None):
     """
-    Store a string in a temporary file, to pass on to a third-party shell command as a file parameter
-    Returns the file path string
+    Store a string in a temporary file
     """
-    fd, file_path = mkstemp(text=True)
-    fp = os.fdopen(fd, 'w')
-    fp.write(string)
-    fp.close()
-    yield file_path
-    os.remove(file_path)
+    f = NamedTemporaryFile('w', delete=False, dir=root_dir)
+    f.write(string)
+    f.close()
+    return f.name
 
 
 @contextmanager
-def run_playbook(requirements_path, inventory_str, vars_str, playbook_path, playbook_name, username='root'):
+def create_temp_dir():
     """
-    Runs ansible-playbook in a dedicated venv
+    A context manager that creates a temporary directory, returns it. Directory is deleted upon context manager exit.
+    """
+    try:
+        temp_dir = mkdtemp()
+        yield temp_dir
+    finally:
+        # This is really a "best effort" solution, if something will block the deletion (like file being locked somehow)
+        # this should just move on rather than raise.
+        shutil.rmtree(temp_dir, ignore_errors=True)
```

Comment:
Yes, but errors raised from ``shutil.rmtree`` would stop the cleanup action --- and possibly leave more garbage in the ``/tmp``, which for me would be a step back. ``rmtree`` bails out after first error, but if it ignores the error it could continue. 

 So this really a decision: 'make maters slightly worse --- by leaving more garbage' vs 'make error less obvious'

Possibly we are discussing a very edge case, we could leave it as it is, or just let ``rmtree`` raise the exception, which would remove some error handling code.